src/main/scala/com/newlibertie/pollster/api/v1/PollApi.scala

```diff
@@ -18,74 +21,72 @@ object PollApi {
-          entity(as[String]) {
-            pollDefinition => {
-              val poll = Poll(pollDefinition)
-              Poll.write(poll) match {
-                case Some(id) => complete(s"""{"id":"${id}"}""")
-                case _ => complete(s"""{"id":""}""") //TODO : send 4XX
-              }
-
-            }
-          }
-        } ~
-        put {
-          parameters("id") { (id: String) =>
-            complete(
-              HttpEntity(
-                ContentTypes.`application/json`,
-                "{}"
-              )
-            )
-          }
-        } ~
-        delete {
-          parameters("id") { (id: String) =>
-            complete(
-              HttpEntity(
-                ContentTypes.`application/json`,
-                "{}"
-              )
-            )
+  lazy val routes: Route = path("poll") {
+    get {
+      parameters("id") { id: String =>
+        val pollStr = DataAdapter.getPoll(id)
+        pollStr match {
+          case poll: String =>
+            complete(HttpResponse(entity = poll))
+          case -1 => complete(StatusCodes.NotFound)
+          case _ => complete(StatusCodes.BadRequest)
+        }
+      }
+    } ~
+    post { // Create a Poll
+      entity(as[String]) {
+        pollDefinition => {
+          val poll = Poll(pollDefinition)
+          Poll.write(poll) match {
+            case Some(id) => complete(StatusCodes.OK, List(`Content-Type`(`text/plain(UTF-8)`)), "id")
+            case _ => complete(StatusCodes.NotFound)
           }
         }
-      } ~
-      path( "closePoll") {
-        get {
-          parameters("id") { (id: String) =>
-            complete(
-              HttpEntity(
-                ContentTypes.`application/json`,
-                "{}"
-              )
+      }
+    } ~
+    put {
+      parameters("id") { id: String =>
+        complete(
+          HttpEntity(
+            ContentTypes.`application/json`,
+            "{}"
+          )
+        )
+      }
+    } ~
+    delete {
+      parameters("id") { id: String =>
+        complete(
+          HttpEntity(
+            ContentTypes.`application/json`,
+            "{}"
+          )
+        )
+      }
+    } ~
+    path("closePoll") {
+      get {
+        parameters("id") { id: String =>
+          complete(
+            HttpEntity(
+              ContentTypes.`application/json`,
+              "{}"
             )
-          }
+          )
         }
-      } ~
-      path( "computeResults") {
-        get {
-          parameters("id") { (id: String) =>
-            complete(
-              HttpEntity(
-                ContentTypes.`application/json`,
-                "{}"
-              )
+      }
+    } ~
+    path("computeResults") {
+      get {
+        parameters("id") { id: String =>
+          complete(
+            HttpEntity(
+              ContentTypes.`application/json`,
+              "{}"
             )
-          }
+          )
         }
       }
     }
-}
+  }
+}
```

Comment:
still seems to have a huge white space change in this file.  perhaps you can ask your editor to not always change whitespace